sample-apps/react/egress-composite/src/CompositeApp.tsx

```diff
@@ -19,14 +19,22 @@ export const CompositeApp = () => {
   if (config.baseURL) {
     options.baseURL = config.baseURL;
   }
-  const client = useCreateStreamVideoClient({
-    apiKey: config.apiKey,
-    tokenOrProvider: config.token,
-    options,
-    user: {
-      id: config.userId,
-    },
-  });
+  const [client] = useState<StreamVideoClient>(
+    () => new StreamVideoClient(config.apiKey, options),
+  );
+
+  useEffect(() => {
+    client.connectUser(
+      {
+        id: config.userId,
+      },
+      config.token,
+    );
+
+    return () => {
+      client.disconnectUser();
+    };
+  }, [client, config.token]);
```

Comment:
```suggestion
  }, [client, config.userId, config.token]);
```

I think `userId` should be provided as a dependency here.